sync/sync.go

```diff
@@ -334,3 +345,18 @@ func (s *Syncer[H]) requestHeaders(
 	}
 	return nil
 }
+
+// verifyHeaders verifies given headers against the store's head.
+func (s *Syncer[H]) verifyHeaders(ctx context.Context, headers []H) ([]H, error) {
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	verified, err := header.VerifyRange(head, headers)
+	if err != nil {
+		return nil, err
```

Comment:
Should we inform that this is a verification error? I mean `fmt.Errof(...)` (not sure should we do for `return` above.

Reply:
yea we can wrap in L300

Reply:
L300 and this func is gonna be removed once based on #269. I did my best to explain this in the large comment.

Reply:
Basically this PR is gonna be just VerifyRanges in local exchange + removal of verification in the Store